src/components/KDropdownMenu/KDropdownItem.vue

```diff
@@ -123,6 +94,60 @@ const handleClick = (event: Event): void => {
     emit('change', props.item)
   }
 }
+
+const preventAndStopDefault = (event: Event): void => {
+  event.preventDefault()
+  event.stopPropagation()
+}
+
+const componentType = computed((): DropdownItemRenderedType => {
+  let result: DropdownItemRenderedType = 'default'
+
+  if (type.value === 'link' && !!to.value && !!props.disabled) {
+    result = 'link'
+  } else if (type.value === 'link' && to.value) {
+    result = 'router-link'
+  } else if (type.value === 'button') {
+    result = 'button'
+  }
+
+  return result
+})
+
+const availableComponents = computed((): DropdownItemRenderedRecord => ({
+  link: {
+    tag: 'a',
+    onClick: preventAndStopDefault,
+    attrs: {
+      class: `k-dropdown-item-trigger ${props.disabled ? 'disabled' : ''}`,
+      href: '#',
+    },
+  },
+  'router-link': {
+    tag: 'router-link',
+    onClick: handleClick,
+    attrs: {
+      class: `k-dropdown-item-trigger ${props.disabled ? 'disabled' : ''}`,
+      to: !props.disabled ? to.value : routePath.value,
+    },
+  },
+  button: {
+    tag: 'KButton',
+    onClick: handleClick,
+    attrs: {
+      class: 'k-dropdown-item-trigger btn-link k-button non-visual-button',
+      disabled: props.disabled,
+      isRounded: false,
+    },
+  },
+  default: {
+    tag: 'div',
+    onClick: undefined,
```

Comment:
nit: if `onClick` is undefined I believe the property here can just be omitted

Reply:
fixed